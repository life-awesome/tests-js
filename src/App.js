import React, {useEffect, useState} from 'react';

const App = () => {
    const [data, setState] = useState(null)
    const [toggle, setToggle] = useState(false)
    const [value, setValue] = useState('')
    const toggleHandler = () => setToggle(p => !p)
    useEffect(() => {
        setTimeout(() => {
            setState({})
        }, 100)
    }, [])
    return (
        <div>
            <div data-testid={'value'}>{value}</div>
            {data && <div>data</div>}
            {toggle && <div data-testid={'menu'}>menu</div>}
            <h1>Hello world</h1>
            <button data-testid="toggle-btn" onClick={toggleHandler}>click me</button>
            <input type="text" placeholder="input-text" value={value} onChange={(e) => setValue(e.target.value)}/>
        </div>
    );
};

export default App;