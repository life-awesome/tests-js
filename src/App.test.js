import {fireEvent, render, screen, waitFor} from '@testing-library/react';
import App from './App';


describe('init', () => {
    test('renders learn react link', () => {
        render(<App/>);
        const linkElement = screen.getByText(/Hello world/i);
        const buttonElement = screen.getByRole('button')
        const inputElement = screen.queryAllByPlaceholderText(/input-text/i)
        expect(linkElement).toBeInTheDocument();
        expect(buttonElement).toBeInTheDocument()
        expect(inputElement).toMatchSnapshot()
    })
    test('APP TEST', async () => {
        render(<App/>)
        const helloWorldElement = await screen.findByText(/data/i);
        expect(helloWorldElement).toBeInTheDocument()
        screen.debug()
    });
    test('toggle handler in component', () => {
        render(<App/>)
        const btn = screen.getByTestId('toggle-btn')
        expect(screen.queryByTestId('menu')).toBeNull()
        fireEvent.click(btn)
        expect(screen.queryByTestId('menu')).toBeInTheDocument()
        fireEvent.click(btn)
        expect(screen.queryByTestId('menu')).toBeNull()
    })
    test('input handler in component', () => {
        render(<App/>)
        const input = screen.getByPlaceholderText(/input-text/i)
        expect(screen.queryByTestId('value')).toContainHTML('')
        fireEvent.input(input, {
            target: {
                value: '12345'
            }
        })
        expect(screen.queryByTestId('value')).toContainHTML('12345')
    })
});
