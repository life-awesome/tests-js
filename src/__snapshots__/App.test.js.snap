// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init renders learn react link 1`] = `
Array [
  <input
    placeholder="input-text"
    type="text"
    value=""
  />,
]
`;
